Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -769,8 +766,11 @@ ShaderCode GeneratePixelShaderCode(APIType api_type, const ShaderHostConfig& hos
     // out.SetConstantsUsed(C_PLIGHT_COLORS, C_PLIGHT_COLORS+7); // TODO: Can be optimized further
     // out.SetConstantsUsed(C_PLIGHTS, C_PLIGHTS+31); // TODO: Can be optimized further
     // out.SetConstantsUsed(C_PMATERIALS, C_PMATERIALS+3);
-    GenerateLightingShaderCode(out, uid_data->lighting, uid_data->components << VB_COL_SHIFT,
-                               "colors_", "col");
+    GenerateLightingShaderCode(out, uid_data->lighting, "colors_", "col");
+    if (uid_data->numColorChans == 0)
+      out.Write("col0 = float4(0.0, 0.0, 0.0, 0.0);\n");
+    if (uid_data->numColorChans <= 1)
+      out.Write("col1 = float4(0.0, 0.0, 0.0, 0.0);\n");
```

Comment:
These lines still use `out.Write`.  I didn't change them because there are a few other lines in this file that also use it instead of `WriteFmt`, and because they don't have any format arguments.  If these should be changed, I'll also change the similar calls in the file.

Reply:
They should be changed, given those two remaining write calls will be migrated in #9197. Tthey were accidentally overlooked